packages/htmlbars-compiler/tests/html-compiler-test.js

```diff
@@ -796,6 +785,15 @@ test('Components - Called as helpers', function () {
   compilesTo('a<x-append text="d{{bar}}">b{{baz}}</x-append>f','abcdef', object);
 });
 
+// IE8 doesn't correctly handle these html strings, so in that case we skip them
+var innerHTMLSupportsCustomTags = (function() {
+  var div = document.createElement('div');
+  div.innerHTML = '<x-bar></x-bar>';
+  return normalizeInnerHTML(div.innerHTML) === '<x-bar></x-bar>';
+})();
```

Comment:
done (except in the test-helper, where it basically all feature detections and fixes for them)